src/couch_log_lager.erl

```diff
@@ -64,10 +64,9 @@ emergency(Fmt, Args) ->
 
 -spec set_level(atom()) -> ok.
 set_level(Level) ->
-    {ok, Handlers} = application:get_env(lager, handlers),
-    lists:foreach(fun({Handler, _}) ->
-        lager:set_loglevel(Handler, Level)
-    end, Handlers).
+    Handlers = gen_event:which_handlers(lager_event),
+    [ok = lager:set_loglevel(Handler, Level) || Handler <- Handlers],
```

Comment:
I'd suggest to keep foreach here, it's more idiomatically correct and you wouldn't need an explicit `ok` at the end.